### PR TITLE
(GH-308) Genericize PCT install methods and files

### PR DESCRIPTION
**Pull request by petergmurphy**
Overview of PR:

- Refactored code pertaining to `pct install` logic to make it more generic so that it can be used by the PRM project as well.
- Refactored `install_tests.go` unit tests, so they use the mock struct within `install_config.go` and added `pct_config_processor.go` with appropriate unit tests to process PCT config files and create namespaced directory structures for the templates.
- Provided suitable mocking for `exec_runner`.

### CODE REVIEW

File: internal/pkg/pct/pct_test.go [hunk 1]
```diff
@@ -777,8 +798,8 @@ template:
 // 	type args struct {
 // 		info         pct.DeployInfo
 // 		configFile   string
-// 		templateFile pct.PuppetContentTemplateFileInfo
-// 		tmpl         pct.PuppetContentTemplate
+// 		templateFile pct.pct.PuppetContentTemplateFileInfo
+// 		tmpl         pct.pct.PuppetContentTemplate
```
Comment (da-ar) [inline internal/pkg/pct/pct_test.go:802]: remove these `pct.` changes throughout this commented code

File: internal/pkg/pct_install/pct_install.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		return "", err
+	}
+
+	namespacePath := filepath.Join(targetDir, info.Template.Author, info.Template.Id, info.Template.Version)
+
+	// finally move to the full path
+	err = p.AFS.Rename(untarPath, namespacePath)
+	// unable to check for this specific error as windows may instead return `access denied`
+	// if err != nil && os.IsExist(err) {
```
Comment (da-ar) [inline internal/pkg/pct_install/pct_install.go:67]: remove

**Review by da-ar — COMMENTED** [review]
A couple of minor code removals. 
This will need eyes on from the rest of the team as I have made  a number of changes on this PR myself.

File: pkg/mock/install.go [hunk 3]
```diff
@@ -38,3 +38,8 @@ func (p *PctInstaller) InstallClone(gitUri, targetDir, tempDir string, force boo
 
 	return filepath.Clean("/unit/test/path"), nil
 }
+
+func (p *PctInstaller) ProcessConfig(sourceDir, targetDir string, force bool) (string, error) {
+	// WIP
```
[the remark this answers is not in the document]
Comment (petergmurphy) [reply pkg/mock/install.go:43]: I don't believe that is needed anymore.

**Review by sanfrancrisko — COMMENTED** [review]
Great job on this 👍

File: pkg/config/config.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package config
+
+type ConfigNamespaceInfo struct {
```
[the remark this answers is not in the document]
Comment (petergmurphy) [reply pkg/config/config.go:3]: I think it's now redundant, I'll remove it.

**Review by michaeltlombardi — APPROVED** [review]
LGTM! 🎉

**Pull request merged**